recipes/libgd/all/conanfile.py

```diff
@@ -0,0 +1,76 @@
+import os
+from conans import ConanFile, tools, CMake
+
+
+class LibgdConan(ConanFile):
+    name = "libgd"
+    license = "https://github.com/libgd/libgd/blob/master/COPYING"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = ("GD is an open source code library for the dynamic "
+                   "creation of images by programmers.")
+    topics = ("images", "graphics")
+    settings = "os", "compiler", "build_type", "arch"
+    homepage = "https://libgd.github.io"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    exports_sources = "CMakeLists.txt", "patches/**"
+    generators = "cmake"
+    requires = "zlib/1.2.11"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def _patch(self):
+        cmakelists = os.path.join(self._source_subfolder, "CMakeLists.txt")
+        tools.replace_in_file(cmakelists, "${CMAKE_SOURCE_DIR}", "${CMAKE_CURRENT_SOURCE_DIR}")
+        for patch in self.conan_data["patches"][self.version]:
```

Comment:
```suggestion
        for patch in self.conan_data.get("patches", {}).get(self.version, []):
```

if in the future you don't a patch, so you don't to check before applying